gdx/src/com/badlogic/gdx/maps/MapLayer.java

```diff
@@ -46,14 +49,32 @@ public void setName (String name) {
 
 	/** @return layer's opacity */
 	public float getOpacity () {
-		return opacity;
+		if (parent != null)
+			return opacity * parent.getOpacity();
+		else
+			return opacity;
 	}
 
 	/** @param opacity new opacity for the layer */
 	public void setOpacity (float opacity) {
 		this.opacity = opacity;
 	}
 
+	/** @return layer's tint color */
+	public Color getTintColor () {
+		if (parent != null) {
+			tempColor.set(tintColor);
+			return tempColor.mul(parent.getTintColor());
```

Comment:
Thanks for taking the time to check this out. It's been a year and a half since I last looked, so I'm giving the code a deeper once over. And with some recent changes I think I'm going to also add in changes to the test maps as well since I should probably add in the Image Layer tinting on the other maps as well.

BUT in reference to your suggestions these all are great. I hope I am following you correctly.
So more in line with something like this for example:
```
	/** @return layer's tint color combined with the parent's tint color */
	public Color getCombinedTintColor(){
		 if (parent != null) {
			  tempColor.set(tintColor);
			  return tempColor.mul(parent.getCombinedTintColor()).cpy();
		 } else {
			  return tintColor.cpy();
		 }
	}

	/** @return layer's tint color */
	public Color getTintColor () {
			return tintColor.cpy();
	}

	/** @param tintColor new tint color for the layer */
	public void setTintColor (Color tintColor) {
		this.tintColor.set(tintColor);
	}

```
Since our worry would be people messing with the tintcolor. Would using Color's .cpy() method be an acceptable solution to lock it down? At most each layer's tint should only be called once per frame.
As for getting the combination I wasn't sure if getCombinedTintColor() would be an acceptable name, since getCombinedParentTintColor() or getCombinedTintWithParentColor(), might be a bit wordy?

Either way I'll gladly take any suggestions and get to work on this and some test map changes a bit later today.

Reply:
I think you'd be fine not cpy()-ing colors here. If it's understood that getTintColor() returns the direct Color being modified, that doesn't need to cpy(). I think since the combined color is always a temporary color, you could change it to:
```java
	/** @return layer's tint color combined with the parent's tint color */
	public Color getCombinedTintColor(){
		 if (parent != null) {
			  // This assumes there's a way to get the combined tint color on parent; you might use the parent's normal color if there's nothing it would combine with.
			  return tempColor.set(tintColor).mul(parent.getCombinedTintColor());
		 } else {
			  return tempColor.set(tintColor);
		 }
	}
```

I'd consider documenting getCombinedTintColor() as returning a temporary Color that should have its value set somewhere else, such as with `Color.set(Color)`, if you need to keep it, since it gets reused. I think it's a good name, it describes what it is. I'm just not sure we... even need to provide the combined color... map renderers could be doing some of this already, at least for alpha. In scene2d.ui, there's no way to get the combined color including parentAlpha; the parentAlpha is provided where it can change a color. Maybe the tint should just be considered by map renderers as part of what they process, but that might not be great with multiple layer tints.